Makefile.include

```diff
@@ -30,6 +30,7 @@ BINDIR         ?= $(BINDIRBASE)/$(BOARD)
 PKGDIRBASE     ?= $(BINDIRBASE)/pkg/$(BOARD)
 DLCACHE        ?= $(RIOTTOOLS)/dlcache/dlcache.sh
 DLCACHE_DIR    ?= $(RIOTBASE)/.dlcache
+BUILD_FILES    ?= $(ELFFILE) $(FLASHFILE)
```

Comment:
I am not sure about defining it as `?=` for that case.

If an application `Makefile` defines it it would overwrite this value and is somehow changing the behavior. As the value must be defined before defining `link`, it cannot be modified after including `Makefile.include` so an application would need to re-add everything. (`ELFFILE` would be generated by `print-size` anyway, but it would be because of black magic)

If you really want to have the `?=` functionality, I would split in two variables, `BUILD_FILES_DEFAULT ?= $(ELFFILE) $(FLASHFILE)` and `BUILD_FILES += $(BUILD_FILES_DEFAULT)`. Not sure if it is currently required though.

For its definition location, the variables at the top are more "directories" related ones. It think it could go at the same place as the definition of `FLASHFILE` (currently 440).

Reply:
addressed